File: src/main/java/com/justice/shopmanagement/viewmodel/GoodsViewModel.java
package com.justice.shopmanagement.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.justice.shopmanagement.model.Goods;
import com.justice.shopmanagement.repository.GoodsRepository;

import java.util.List;

public class GoodsViewModel extends AndroidViewModel {
    private GoodsRepository repository;
    private LiveData<List<Goods>> allGoods;

    public GoodsViewModel(@NonNull Application application) {
        super(application);
        repository = new GoodsRepository(application);
        allGoods = repository.getAllGoods();
    }

    public void insert(Goods goods) {
        repository.insert(goods);
    }

    public void update(Goods goods) {
        repository.update(goods);
    }

    public void delete(Goods goods) {
        repository.delete(goods);
    }

    public void deleteAllGoods() {
        repository.deleteAllGoods();
    }

    public LiveData<List<Goods>> getAllGoods() {
        return allGoods;
    }
}

File: app/src/main/java/com/justice/shopmanagement/viewmodel/GoodsViewModel.kt
package com.justice.shopmanagement.viewmodel

import android.util.Log
import androidx.hilt.lifecycle.ViewModelInject
import com.justice.shopmanagement.repository.ShopRepository
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.justice.shopmanagement.model.Goods
import kotlinx.coroutines.launch

class GoodsViewModel @ViewModelInject constructor(private val repository: ShopRepository) :
    ViewModel() {
    private val TAG = "GoodsViewModel"

    val allGoods: MutableLiveData<List<Goods>> = MutableLiveData()

    fun insert(goods: Goods?) {
        try {
            viewModelScope.launch {
                Log.d(TAG, "insert: insert in progress...")

                val resp = repository.insert(goods)
                if (resp.isSuccessful) {
                    allGoods.value = resp.body()
                    Log.d(TAG, "insert: resp:${resp.body()}")

                } else {
                    Log.e(TAG, "insert: ${resp.errorBody()}")
                }
                Log.d(TAG, "insert: insert finished")
            }
        } catch (e: Exception) {
            Log.e(TAG, "insert: ", e)
        }

    }

    fun update(goods: Goods?) {

        //repository.update(goods);
    }

    fun delete(goods: Goods?) {

        // repository.delete(goods);
    }

    fun deleteAllGoods() {

        //repository.deleteAllGoods();
    }


}